File: solutions/fibonacci.py
def fibonacci(num):

    if num < 2:
        return num
    
    last_two = (0, 1)

    for i in range(0, num - 1, 1):
        sum = last_two[0] + last_two[1]
        last_two = (last_two[1], sum)

    return last_two[1]

if __name__ == "__main__":
  
  print("Expecting: 0")
  print(fibonacci(0))

  print("")

  print("Expecting: 1")
  print(fibonacci(2))

  print("")

  print("Expecting: 55")
  print(fibonacci(10))

  print("")

  print("Expecting: 1")
  print(fibonacci(1))

  print("")

  print("Expecting: 6765")
  print(fibonacci(20))

File: javascript/fibonacci.js
function fibonacci(num) {
  // type your code here
  
  let value1 = 0;
  let value2 = 1;

  let sum = 0;
  for (let i = 1; i <= num; i++){
        sum = value1 + value2;
        value1 = value2;
        value2 = sum;
     
  }

  return value1
}

if (require.main === module) {
  // add your own tests in here

  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(1));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));
}

module.exports = fibonacci;

// Please add your pseudocode to this file
// And a written explanation of your solution
;